### PR TITLE
Handle unpropagated `this.parse` calls

### PR DESCRIPTION
**Pull request by KrisXV**
(also fixes the build)

### CODE REVIEW

File: server/chat-commands/core.ts [hunk 1]
```diff
@@ -414,11 +414,11 @@ export const commands: ChatCommands = {
 			return this.sendReply(this.tr("What?! How are you not more excited to battle?! Try /battle! to show me you're ready."));
 		}
 		if (!target) target = "randombattle";
-		return this.parse(`/search ${target}`);
+		return void this.parse(`/search ${target}`);
```
Comment (Zarel) [inline server/chat-commands/core.ts:417]: Err, `return` already correctly propagates `this.parse`; the `void` is unnecessary.

Comment (KrisXV) [reply server/chat-commands/core.ts:417]: idk how i messed that up so bad, oops

**Comment by Zarel** [comment]
For some reason the build errors didn't show up for me when I did `npm test` or `npm run lint`. I guess a bug in eslintcache? I dealt with similar problems with other `no-floating-promises` errors randomly appearing and disappearing. I guess I'll remove the return value for now.

File: server/chat-commands/core.ts (hunk 1 above)
Comment (Zarel) [reply server/chat-commands/core.ts:417]: The `void` is actually bad because it stops the promise from being propagated like it should be.

**Pull request closed without merging**